packages/ra-core/src/reducer/admin/resource/data.js

```diff
@@ -60,40 +57,43 @@ const addRecords = addRecordsFactory(getFetchedAt);
 const initialState = {};
 Object.defineProperty(initialState, 'fetchedAt', { value: {} }); // non enumerable by default
 
-export default (previousState = initialState, { type, payload, meta }) => {
-    if (type === CRUD_UPDATE_OPTIMISTIC) {
-        const updatedRecord = { ...previousState[payload.id], ...payload.data };
-        return addRecords([updatedRecord], previousState);
-    }
-    if (type === CRUD_UPDATE_MANY_OPTIMISTIC) {
-        const updatedRecords = payload.ids
-            .reduce((records, id) => records.concat(previousState[id]), [])
-            .map(record => ({ ...record, ...payload.data }));
-        return addRecords(updatedRecords, previousState);
-    }
-    if (type === CRUD_DELETE_OPTIMISTIC) {
-        const { [payload.id]: removed, ...newState } = previousState;
+export default (previousState = initialState, { payload, meta }) => {
+    if (meta && meta.optimistic) {
+        if (meta.fetch === UPDATE) {
```

Comment:
Add a getUpdatedRecords function to get the updatedResults

Reply:
I don't understand this one.

Reply:
I mean extract the logic to get the updatedResults in a separate function.
This way you can call addRecords a single time at the end.

Reply:
Not all cases require to call `addRecords`, I think it is more readable this way